packages/snaps-utils/src/ui.ts

```diff
@@ -17,35 +18,39 @@ export function validateTextLinks(
   text: string,
   isOnPhishingList: (url: string) => boolean,
 ) {
-  const links = text.match(LINK_REGEX);
-  if (links) {
-    links.forEach((link) => {
-      try {
-        const url = new URL(link);
-        assert(
-          ALLOWED_PROTOCOLS.includes(url.protocol),
-          `Protocol must be one of: ${ALLOWED_PROTOCOLS.join(', ')}.`,
-        );
+  const matches = text.matchAll(MARKDOWN_LINK_REGEX);
+  if (matches) {
+    for (const { groups } of matches) {
+      const link = groups?.url;
+
+      if (link) {
```

Comment:
Could we rewrite to reduce the nesting? It's a bit difficult to read now.

Can probably be done with 
```js
if (!link) {
  continue;
}
```

or a filter on the `matches`?